sdk/js/src/Client.ts

```diff
@@ -271,6 +273,25 @@ export class DVCClient implements Client {
             this.logger.error('Streaming Connection: Unparseable message', e)
         }
     }
+
+    private registerVisibilityChangeHandler() {
+        if (typeof document !== 'undefined') {
+            this.pageVisibilityHandler = () => {
+                if (document.visibilityState === 'visible') {
+                    this.logger.debug('Page became visible, refetching config')
+                    this.refetchConfig().catch((e) => {
+                        this.logger.error('Failed to refetch config', e)
+                    })
+                    this.streamingConnection?.reopen()
+                } else {
+                    this.logger.debug('Page is not visible, closing streaming connection')
+                    this.streamingConnection?.close()
+                }
+            }
+
+            document.addEventListener?.('visibilitychange', this.pageVisibilityHandler)
```

Comment:
should we also remove this event listener when the page closes?

Reply:
you mean like when the page is actually closed? there can't be any event handlers after the page has closed anyway

Reply:
he does that in the next PR lol